### PR TITLE
feat: Implements tls in the charm using vault_tls lib

### PR DESCRIPTION
**Pull request by saltiyazan**
# Description

 - Implements tls_certificates and cewrtificate_transfer interfaces in the charm using vault_tls lib
 - Updates requirements
 - Updates tests
 - Uses a pyproject.toml that matches the one the k8s charm (similar PR in k8s for that [here](https://github.com/canonical/vault-k8s-operator/pull/176)).

# Checklist:

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that validate the behaviour of the software
- [x] I validated that new and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/integration/test_integration.py [hunk 1]
```diff
@@ -88,7 +88,7 @@ async def test_given_charm_deployed_and_active_when_vault_status_checked_then_va
         timeout=1000,
     )
     unit_ip = ops_test.model.units.get(f"{APP_NAME}/0").public_address
-    vault_endpoint = f"http://{unit_ip}:8200"
+    vault_endpoint = f"https://{unit_ip}:8200"
     # TODO: Use certs in "verify" when added in charm.
     client = hvac.Client(url=vault_endpoint, verify=False)
```
Comment (gruyaume) [inline tests/integration/test_integration.py:93]: We shouldn't need the `verify=False` now

File: charmcraft.yaml [hunk 2]
```diff
@@ -13,4 +13,10 @@ bases:
 parts:
   charm:
     charm-binary-python-packages:
+      - cryptography
```
Comment (gruyaume) [inline charmcraft.yaml:16]: No need to add this here since we added `rustc` and the others, we should be able to build it.

**Review by gruyaume — CHANGES_REQUESTED** [review]

File: tests/unit/test_charm.py [hunk 3]
```diff
@@ -114,6 +99,24 @@ def _set_other_node_api_address_in_peer_relation(self, relation_id: int, unit_na
             key_values=key_values,
         )
 
+    def _set_ca_certificate_secret(self, private_key: str, certificate: str) -> None:
+        """Set the certificate secret in the peer relation."""
```
Comment (Kayra1) [inline tests/unit/test_charm.py:103]: Did this docstring come from the copy paste? Should change it here and there if so.

**Review by Kayra1 — APPROVED** [review]
Looks like we didn't have to duplicate pretty much anything related to TLS. Great job.

**Review by gruyaume — APPROVED** [review]

**Pull request merged**